### ISSUE TITLE
feature: dynamic schema/server response viewer

### ISSUE THREAD

**Issue by JonKohler** [swagger-api/swagger-ui#4372]
| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  feature
| Which Swagger/OpenAPI version?  | 2.0
| Which Swagger-UI version?       | 3.latest
| How did you install Swagger-UI? | n/a
| Which browser & version?        | n/a
| Which operating system?         | n/a

### Current Behavior
Currently, the "Example Value" and "Server response" display section in swagger-ui 3.x are just static canvas, which works very well for small examples, for both schema, intended response (examples), and server responses. 

For API's that have large inputs and/or large outputs, or for users with relatively low-resolution screens, this doesn't scale well for them

### Possible Solution
We've been working on a react based replacement for our swagger-ui 2.1.4 implementation within our product; however, we're currently changing direction to baseline this as an upgrade from 2.1.4 to 3.x, and would like to port in some of the features that we've added.

Specifically for this feature request, we've built a dynamic schema/server response viewer, that has the following characteristics: 

1. Great line/section copy to clipboard method, with both the inline copy (which copies either the single line or if done at a tree head, copies that section of the tree). Also has a fancy "copy" animation checkmark
2. Great formatting and stylization, Line numbers, collapsible trees, nicely stylized like common editors (like sublime text, etc)
3. Ability to take response viewer into a full-screen mode with a single click, to really dig into big responses easily, WITHOUT taking them to an external JSON parser or external tool
4. Great "show" function, where it unfolds the JSON just a little bit and tells you how many subitems are in the tree. This is VERY useful for API's that have VERY big schemas/server responses

This would replace the "Example Value" display section in swagger-ui 3 for schema and responses, which is just a static canvas now, AND should also replace the "Response body" section when using the Try It Out method.

### Screenshots
I've included a few screenshots from the internal code that we have.

Posting this here to garner feedback, so that we can shape this for optimal upstream acceptance once we're done.

<img width="1138" alt="screenshot 2018-03-22 02 33 45" src="https://user-images.githubusercontent.com/21041006/37928093-bfb7f2c6-3109-11e8-84b0-8c294aded588.png">
<img width="682" alt="screenshot 2018-03-22 02 36 33" src="https://user-images.githubusercontent.com/21041006/37928119-cc3311ac-3109-11e8-805a-14e610c24caa.png">
<img width="714" alt="screenshot 2018-03-22 02 36 42" src="https://user-images.githubusercontent.com/21041006/37928146-de7b9b22-3109-11e8-81cf-8bfe670ae0ba.png">
<img width="1134" alt="screenshot 2018-03-22 02 42 39" src="https://user-images.githubusercontent.com/21041006/37928156-e4cd6604-3109-11e8-87ec-d799b3b70ee3.png">

**Comment by cirojr** [comment]
That's exactly what I was thinking to be a good improvement. Well done.
Do you plan to do a PR, @JonKohler?

**Comment by JonKohler** [comment]
Thanks @cirojr. We've taken one step back to "look around" again at whether we want to do this (in general), or if we're going to stick with swagger-ui, or go to another framework for our product. Its nothing personal if we end up moving away, but that would mean that we would not put up a PR for this.

For sure we know that our company doesn't want to be in the business of writing and maintaining (yet another) swagger/openapi documentation engine, so its just a question if we "upgrade" from swagger-ui 2.x or go with another solution. I've got a developer tasks with this investigation now, and we're trying to get to the bottom of it this week. 

I'll let everyone know where we end up going, and close out this PR if we're going to abandon ship, so that people aren't left expecting this.

**Comment by heldersepu** [comment]
After using the UI for while I really don't think the additional bells & whistles are a good thing.
And for " VERY big schemas/server responses " I'm sure that performance will be horrible, try this one:
http://swagger-net-test.azurewebsites.net/swagger/ui/index?filter=Huge#/HugeResponse/HugeResponse_Get

**Comment by JonKohler** [comment]
It's not about performance, its about UX/DX usability. This is one of the reasons why we're looking at other options as I noted above, as this is not the only example where swagger-ui 3.x (as is) really doesn't scale well with large APIs, with large responses, lots of endpoints, etc. Again, performance is fine, usability is not.

**Comment by heldersepu** [comment]
@JonKohler It's not about performance for you...
But performance is a concern for many of us, do you have a working prototype I can test?

**Comment by JonKohler** [comment]
No, we dont have a working prototype available, we were just hacking it together in react when we decided we need to take a step back and reflect on our requirements, and the user experience we're trying to provide in our product. Should have a decision on that today.

**Comment by heldersepu** [comment]
@JonKohler  ...Interesting, I do a lot of hacking myself too.
but I do keep it on GitHub (fork it & branch it) that way my work has potential to help someone else, as long as it is on GitHub it is easy to get my half hacked code and take it to the next level

Also keep in mind that 3.x can be easily customized using the Plugin Api:
https://github.com/swagger-api/swagger-ui/blob/master/docs/customization/plugin-api.md
Looks that what you need is to Wrap the [HighlightCode ](https://github.com/swagger-api/swagger-ui/blob/2795518340b6c424dd96e14e61a47588609474de/src/core/components/highlight-code.jsx#L6)component

**Comment by webron** [comment]
@JonKohler - Sorry for the delayed reply, I was out for a while, and emails kept piling on.

Regardless of your decision (which I respect), I'd appreciate hearing the challenges you've encountered with regard to: 
> as this is not the only example where swagger-ui 3.x (as is) really doesn't scale well with large APIs, with large responses, lots of endpoints, etc. Again, performance is fine, usability is not.

We're regularly looking to improve the UI, and means addressing both simple and complex APIs, different requirements and needs. As you can imagine, addressing _everyone's_ needs is never going to be possible, but by understanding. If you're open for a short call, please let me know (email is in my profile), or if you prefer putting the details here, that's fine too - all assuming you can make the time to provide the feedback.

I'm not addressing (yet) the suggested features here as I'd like to spend some more time understanding the proposals.

**Comment by JonKohler** [comment]
hey @webron I'd appreciate that, thank you so much! I'll reach out right now, and loop in both our product manager and developer responsible.

**Comment by webron** [comment]
@JonKohler - thanks! That would be great help for the project :)
